shared/src/main/java/uk/gov/di/authentication/shared/services/DynamoService.java

```diff
@@ -532,11 +533,11 @@ public UserProfile getUserProfileFromSubject(String subject) {
                 QueryEnhancedRequest.builder().consistentRead(false).queryConditional(q).build();
         Optional<UserProfile> userProfile =
                 subjectIDIndex.query(queryEnhancedRequest).stream()
-                        .limit(1)
-                        .map(t -> t.items().get(0))
-                        .findFirst();
+                        .findFirst()
+                        .flatMap(page -> page.items().stream().findFirst());
         if (userProfile.isEmpty()) {
-            throw new RuntimeException("No userCredentials found with query search");
+            throw new UserNotFoundBySubjectIdRuntimeException(
+                    "No user profile found with query search");
```

Comment:
Might be worth duplicating this method and making the exception a checked exception to enforce it gets handled and to avoid impacting anything else that already calls it

Reply:
separate method created